lib/main.ts

```diff
@@ -497,19 +499,30 @@ export class Transpiler {
     this.emit(translated);
   }
 
-  static DART_KEYWORDS =
-      ('abstract as assert async await break case catch class const continue ' +
-       'default deferred do dynamic else enum export extends external factory false final ' +
-       'finally for get if implements import in is library new null operator part rethrow return' +
-       ' set static super switch sync this throw true try typedef var void while with yield')
-          .split(/ /);
+  // Dart keyword list copied from
+  // https://www.dartlang.org/docs/dart-up-and-running/ch02.html#keywords
+  // Note the different kinds of keywords defined by the enum; key values
+  // match the footnote numbers given in the docs, or 0 if none given.
+  static DART_KEYWORDS : { [key:string] : DART_KEYWORD_KIND } = {
+    'abstract':1,	'continue':0,	'false':0,	'new':0,	'this':0,
+    'as':1,	'default':0,	'final':0,	'null':0,	'throw':0,
+    'assert':0,	'deferred':1,	'finally':0,	'operator':1,	'true':0,
```

Comment:
As these are enum values, they really should use the enum. It might be easier to actually have two sets and just do a lookup in them, rather than having the enum.

Also please mind the formatting here. I had it as a string with split as I thought that was more readable than a key per line, YMMV.